### PR TITLE
Add support for genome build 38

### PR DESCRIPTION
**Pull request by adrosenbaum**
- Chose which build to use in cli, or in config (defaults to 37)
- remove all hard coding related to build 37 (par regions)
- Add 'profiling'-variants based on what genome build is specified
- Always build variant id excluding possible 'chr' contig notation

### CODE REVIEW

**Comment by adrosenbaum** [comment]
Hi,
@henrikstranneheim and I talked about making loqusdb compatible with the latest genome build. This PR tries to fix this, removing all hard coding to build 37. The profiling-variants now exists as resources in this repo, both for the 37, and 38 build. I also changed so that in case a vcf-file uses 'chr' in the contig names, this prefix is ignored when creating the variant IDs. Let me know if you have any comments or suggestions @moonso

**Comment by moonso** [comment]
Hi @adrosenbaum , thanks for this!
I wonder about all these global parameter stuff, are they really necessary? To me they are mostly confusing. Why not use '37' and '38' everywhere. We know that will not change in many years. I give some suggestions in the code and then you can reason about your approach?

File: loqusdb/build_models/variant.py [hunk 1]
```diff
@@ -5,7 +5,8 @@
 
 from loqusdb.models import Variant
 from loqusdb.exceptions import CaseError
-from loqusdb.constants import (PAR,GENOTYPE_MAP,CHROM_TO_INT)
+from loqusdb.constants import (PAR,GENOTYPE_MAP,CHROM_TO_INT,GRCH37,GRCH38)
```
Comment (moonso) [inline loqusdb/build_models/variant.py]:
```suggestion
from loqusdb.constants import (PAR,GENOTYPE_MAP,CHROM_TO_INT)
```

File: loqusdb/build_models/variant.py [hunk 2]
```diff
@@ -5,7 +5,8 @@
...
+from loqusdb.constants import global_parameters
```
Comment (moonso) [inline loqusdb/build_models/variant.py]:
```suggestion

```

File: loqusdb/build_models/variant.py [hunk 3]
```diff
@@ -25,7 +26,7 @@ def check_par(chrom, pos):
     """
     par = False
 
-    for interval in PAR.get(chrom,[]):
+    for interval in PAR[global_parameters.GENOME_BUILD].get(chrom, []):
```
Comment (moonso) [inline loqusdb/build_models/variant.py]:
```suggestion
    for interval in PAR[build].get(chrom, []):
```
just add `build='37'` to the arguments above

File: loqusdb/commands/cli.py [hunk 4]
```diff
@@ -16,6 +16,8 @@
 from loqusdb.log import LEVELS, init_log
 from loqusdb import __version__
 from loqusdb.plugins import MongoAdapter
+from loqusdb.constants import GRCH37, GRCH38
```
Comment (moonso) [inline loqusdb/commands/cli.py:19]:
```suggestion

```

File: loqusdb/commands/cli.py [hunk 5]
```diff
@@ -16,6 +16,8 @@
...
+from loqusdb.constants import global_parameters
```
Comment (moonso) [inline loqusdb/commands/cli.py]:
```suggestion

```

File: loqusdb/commands/cli.py [hunk 6]
```diff
@@ -56,10 +58,13 @@
                 is_flag=True,
                 help='Used for testing. This will use a mongomock database.'
 )
+@click.option('-g', '--genome-build',
+              type=click.Choice([GRCH37, GRCH38]),
```
Comment (moonso) [inline loqusdb/commands/cli.py:61]:
```suggestion
              type=click.Choice(['37','38']),
```

File: loqusdb/commands/cli.py [hunk 7]
```diff
@@ -109,3 +114,7 @@ def cli(ctx, database, username, password, authdb, port, host, uri, verbose, con
     ctx.obj['host'] = host
     ctx.obj['adapter'] = adapter
     ctx.obj['version'] = __version__
+
+    genome_build = genome_build or configs.get('genome_build') or GRCH37
```
Comment (moonso) [inline loqusdb/commands/cli.py]:
```suggestion
    genome_build = genome_build or configs.get('genome_build') or '37'
```

File: loqusdb/commands/cli.py [hunk 8]
```diff
@@ -109,3 +114,7 @@ def cli(ctx, database, username, password, authdb, port, host, uri, verbose, con
...
+    global_parameters.set_genome_build(genome_build)
```
Comment (moonso) [inline loqusdb/commands/cli.py]:
```suggestion

```

File: loqusdb/commands/cli.py [hunk 9]
```diff
@@ -109,3 +114,7 @@ def cli(ctx, database, username, password, authdb, port, host, uri, verbose, con
...
+    LOG.info("Using build %s", global_parameters.GENOME_BUILD)
```
Comment (moonso) [inline loqusdb/commands/cli.py]:
```suggestion
    LOG.info("Using build %s", genome_build)
```

File: loqusdb/commands/load_profile.py [hunk 10]
```diff
@@ -3,12 +3,12 @@
 import json
 
 from loqusdb.utils.load import load_profile_variants
-
 from loqusdb.utils.profiling import (update_profiles,
                                      profile_stats,
                                      check_duplicates,
                                      get_profiles)
-
+from loqusdb.constants import global_parameters
```
Comment (moonso) [inline loqusdb/commands/load_profile.py]:
```suggestion

```

**Comment by adrosenbaum** [comment]
Hi, I can change this. I set the genome build as a global, to avoid passing this as an argument to all functions, and embedded functions, that needs this information. Agree that it adds some complexity to the code.

**Review by moonso — CHANGES_REQUESTED** [review]
Ok I think I made my point, it is tideous to code on GitHub 😅 . I would very much prefer to pass arguments to functions instead of using globals as much as possible.

File: loqusdb/build_models/variant.py (hunk 1 above)
Comment (adrosenbaum) [reply loqusdb/build_models/variant.py]: Ok, no problems, I'll fix that!

File: loqusdb/build_models/variant.py [hunk 11]
```diff
@@ -24,8 +24,10 @@ def check_par(chrom, pos):
             par(bool)
     """
     par = False
-
-    for interval in PAR.get(chrom,[]):
+    if genome_build is None:
+        genome_build=GRCH37
+    for interval in PAR[genome_build].get(chrom, []):
+        print(interval)
```
Comment (moonso) [inline loqusdb/build_models/variant.py]:
```suggestion

```
:)

**Review by moonso — APPROVED** [review]
I approve but please just remove that print. Thanks @adrosenbaum !

File: loqusdb/build_models/variant.py (hunk 11 above)
Comment (adrosenbaum) [reply loqusdb/build_models/variant.py]: Oops! Will remove!
Thanks for review. We don't need to merge quite yet. I haven't had time to test this on stage, and I will need to wait until hasta on Monday!

**Comment by adrosenbaum** [comment]
Tested on stage, works fine.

**Pull request merged**